src/layouts/Login/components/OtpForm.tsx

```diff
@@ -0,0 +1,104 @@
+import { HStack, FormControl } from "@chakra-ui/react"
+import {
+  Button,
+  FormLabel,
+  Input,
+  FormErrorMessage,
+} from "@opengovsg/design-system-react"
+import { useEffect, useState } from "react"
+import { useForm } from "react-hook-form"
+
+export type OtpProps = {
+  otp: string
+}
+
+interface OtpFormProps {
+  email: string
+  onSubmit: (inputs: OtpProps) => Promise<void>
+  onResendOtp: () => Promise<void>
+  errorMessage: string
+}
+
+const OTP_TIMER_INTERVAL = 60
+
+const validateOtp = (value: string) =>
+  value.length === 6 || "Please enter a 6 digit OTP."
+
+const getOtpResendMessage = (remainingTime: number): string => {
+  if (remainingTime === 0) {
+    return "Resend OTP"
+  }
+
+  return `Resend OTP in ${remainingTime}s`
+}
+
+export const OtpForm = ({
+  email,
+  onSubmit,
+  onResendOtp,
+  errorMessage,
+}: OtpFormProps): JSX.Element => {
+  const [timer, setTimer] = useState(OTP_TIMER_INTERVAL)
+  const { handleSubmit, register, formState, setError } = useForm<OtpProps>({
+    mode: "onBlur",
+  })
+
+  useEffect(() => {
+    if (errorMessage)
+      setError("otp", {
+        type: "server",
+        message: errorMessage,
+      })
+  }, [errorMessage, setError])
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (timer > 0) setTimer(timer - 1)
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [timer])
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <FormControl
+        isInvalid={!!formState.errors.otp}
+        isReadOnly={formState.isSubmitting}
+      >
+        <FormLabel isRequired htmlFor="otp">
+          {`Enter OTP sent to ${email}`}
+        </FormLabel>
+        <Input
+          type="text"
+          maxLength={6}
+          inputMode="numeric"
+          autoComplete="one-time-code"
+          {...register("otp", {
+            pattern: {
+              value: /^[0-9\b]+$/,
```

Comment:
according to regex101, the `\b` matches a backspace character - i assume this is to allow for backspacing right?

if this is so, could we just jot it down as a comment on the top cos it isn't immediately apparent what the `\b` implies

Reply:
It's actually for a word boundary! ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions))

Reply:
til on word boundaries! could we add a short comment explaining that the \b is for word boundaries + link to the mdn article